pkg/controller/runGC.go

```diff
@@ -0,0 +1,154 @@
+package controller
+
+import (
+	"context"
+	"errors"
+	"os"
+	"sort"
+	"strings"
+
+	hephv1 "github.com/dominodatalab/hephaestus/pkg/api/hephaestus/v1"
+	"github.com/dominodatalab/hephaestus/pkg/clientset"
+	"github.com/dominodatalab/hephaestus/pkg/config"
+	"github.com/dominodatalab/hephaestus/pkg/kubernetes"
+	"github.com/go-logr/logr"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8s "k8s.io/client-go/kubernetes"
+	ctrlzap "sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+type ImageBuildGC struct {
+	maxIBRetention int
+	hephClient     clientset.Interface
+	namespaces     []string
+}
+
+func NewImageBuildGC(maxIBRetention int, log logr.Logger, ibNamespaces []string) (*ImageBuildGC, error) {
+	config, err := kubernetes.RestConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	hephClient, _ := clientset.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	var ns []string
+	if len(ibNamespaces) > 0 {
+		log.Info("Image Build GC limiting namespaces", "namespaces", ibNamespaces)
+		ns = ibNamespaces
+	} else {
+		// create k8s client to access all namespaces in the cluster
+		k8sClient, err := k8s.NewForConfig(config)
+		if err != nil {
+			return nil, err
+		}
+		ns, err = getAllNamespaces(k8sClient)
+		if err != nil {
+			log.Info("Unable to access cluster namespaces")
+			return nil, err
```

Comment:
Consider using error wrapping instead of the log + return pattern:
```golang
return nil, fmt.Errorf("Unable to access cluster namespaces: %w", err)
```